src/utils/dexFormater.js

```diff
@@ -85,47 +102,30 @@ const calculateAmountMin = (
   subResult = true,
 ) => {
   const defaultMinPercent = 0.5;
-  const minAmountValue = (Number(minAmountPercent || defaultMinPercent) / 100);
-  const multiplier = new BN((minAmountValue * 1e18).toString());
-  const { div, mod } = amount.mul(multiplier).divmod(new BN(1e18.toString()));
-  const stringValue = mod.toString();
-
-  const integerPart = stringValue.slice(0, -1e18.toString().length) || '0';
-  const decimalPart = stringValue.slice(-1e18.toString().length);
-
-  const dividedValueString = `${integerPart}.${decimalPart}`;
-  const substract = new BN(Number(div) + Number(dividedValueString));
-  const result = subResult
-    ? amount.sub(substract.isZero() ? new BN(2) : substract)
-    : amount.add(substract.isZero() ? new BN(2) : substract);
-  return result;
-};
-
-export const calculateAmountDesiredFormatted = (desiredAmount, decimals) => {
-  const valueString = desiredAmount.toString();
-  const decimalsFromValue = valueString.split('.')[1]?.length || '0';
-  return new BN(valueString.replace('.', ''))
-    .mul(decimals)
-    .div((new BN(10)
-      .pow(new BN(decimalsFromValue))));
+  const minAmountValue = (Number(minAmountPercent || defaultMinPercent) / 100).toString();
+  const lengthDecimalPercent = minAmountValue.split('.')[1].length;
+  const minAmountPercentBN = parseAssets(minAmountValue, lengthDecimalPercent);
+  const decimalsBN = new BN(10).pow(new BN(lengthDecimalPercent));
+  const percentBN = amount.mul(minAmountPercentBN).divn(decimalsBN);
+
+  if (subResult) {
+    const sub = amount.sub(percentBN.isZero() ? new BN(3) : percentBN);
+    return sub.lte(new BN(0)) ? 1 : sub;
+  }
+  return amount.add(percentBN.isZero() ? new BN(3) : percentBN);
```

Comment:
I assume that the `add` case is actually for calculating `amountMax`, not `amountMin`. This doesn't belong in a single function. Call this function (that calculates the slippage) `calculateSlippage` and do a separate `calculateAmountMin` that subs it and `calculateAmountMax` that adds it.

Also it's not a formatter in any way, it's business logic, so doesn't belong in `dexFormater.js`